File: tests/test-clients/java-okhttp-moshi/src/test/java/test_client/CheckClientTest.java
package test_client;

import org.junit.jupiter.api.Test;
import test_client.clients.check.CheckClient;
import test_client.errors.ClientException;

import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
import static org.junit.jupiter.api.Assertions.assertThrows;
import static test_client.Constants.BASE_URL;
import static test_client.Constants.MESSAGE;

public class CheckClientTest {
	private final CheckClient client = new CheckClient(BASE_URL);

	@Test
	public void checkEmpty_doesntThrowException() {
		assertDoesNotThrow(client::checkEmpty);
	}

	@Test
	public void checkEmptyResponse_doesntThrowException() {
		assertDoesNotThrow(() -> client.checkEmptyResponse(MESSAGE));
	}

	@Test
	public void checkForbidden_doesntThrowException() {
		assertThrows(ClientException.class, client::checkForbidden);
	}
}

File: tests/test-clients/kotlin-okhttp-moshi/src/test/kotlin/test_client/CheckClientTest.kt
package test_client

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test
import test_client.clients.check.*
import test_client.errors.*
import test_client.models.*

class CheckClientTest {
    private val client = CheckClient("http://localhost:8081")

    private val intValue: Int = 123
    private val stringValue: String = "the value"
    private val bodyMessage = Message(intValue, stringValue)

    @Test
    fun checkEmpty_doesntThrowException() {
        assertDoesNotThrow(client::checkEmpty)
    }

    @Test
    fun checkEmptyResponse_doesntThrowException() {
        assertDoesNotThrow { client.checkEmptyResponse(bodyMessage) }
    }

    @Test
    fun checkForbidden_throwException() {
        assertThrows(ClientException::class.java) { client.checkForbidden() }
    }
}